python/tvm/octo/octo_model.py

```diff
@@ -0,0 +1,191 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, wrong-import-position
+"""Wrapper class for compiled models."""
+import json
+import tarfile
+import numpy as np
+from pathlib import Path
+from typing import Optional, Union, Dict, Tuple, List
+import tvm
+from tvm import relax
+from tvm.contrib import utils
+
+
+class OctoModel(object):
+    """A compiled model wrapper that provides helpful utilities.
+
+    Parameters
+    ----------
+    exe : Optional[relax.Executable]
+        A compiled executable that can be loaded and run by a relax VM.
+    input_info : Optional[Dict[str, Tuple[List, str]]]
+        Information about the input names, shapes, and types for the VM.
+        Will be loaded from memory if possible.
+    model_path : Optional[Union[str, Path]]
+        The path to a saved OctoModel, one of exe and model_path must
+        be specified.
+    target : Optional[tvm.target.Target]
+        The target being compiled for.
+    """
+
+    def __init__(
+        self,
+        exe: Optional[relax.Executable] = None,
+        input_info: Optional[Dict[str, Tuple[List, str]]] = None,
+        model_path: Optional[Union[str, Path]] = None,
+        target: Optional[tvm.target.Target] = None,
+    ):
+        self.target = target
+
+        if exe is None and model_path is None:
+            raise ValueError("One of vm and model_path must be provided.")
+
+        self._tmp_dir = utils.tempdir()
+
+        if model_path is not None:
+            exe, input_info = self.load(model_path)
+
+        self.dev = tvm.device(self.target.get_target_device_type())
+        self.exe = exe
+        self.input_info = input_info
+
+        # Create a vm from exe.
+        self.vm = relax.VirtualMachine(self.exe, self.dev, profile=True)
+
+    def save(
+        self, model_path: Union[str, Path]
+    ) -> Tuple[relax.Executable, Dict[str, relax.StructInfo]]:
```

Comment:
it looks like nothing is actually returned
```suggestion
    ) -> None:
```